File: src/main/java/fr/spoutnik87/musicbot_rest/repository/PermissionRepository.java
package fr.spoutnik87.musicbot_rest.repository;

import fr.spoutnik87.musicbot_rest.model.Permission;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface PermissionRepository extends JpaRepository<Permission, Long> {

    Optional<Permission> findByValue(String value);

    Optional<Permission> findByUuid(String uuid);
}

File: src/main/kotlin/fr/spoutnik87/musicbot_rest/repository/PermissionRepository.kt
package fr.spoutnik87.musicbot_rest.repository

import fr.spoutnik87.musicbot_rest.model.Permission
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import org.springframework.transaction.annotation.Transactional

@Repository
interface PermissionRepository : JpaRepository<Permission, Long> {
    @Transactional(readOnly = true)
    fun findByUuid(uuid: String): Permission?
    @Transactional(readOnly = true)
    fun findByValue(value: String): Permission?
}